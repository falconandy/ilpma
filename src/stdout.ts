export default class StdOut {
  static write(text: string) {
    process.stdout.write(text);
  }

  static writeLine(text?: string) {
    if (text) {
      process.stdout.write(text);
    }
    process.stdout.write('\n');
  }
}
