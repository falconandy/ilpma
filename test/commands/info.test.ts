import { resetAll, resetMocks } from '../test-helpers/helper.mocks';
import { mockStdOut, verifyStdOut } from './mocks/stdout';

describe(`Test 'info'`, () => {
  const mocks: { [id: string]: jest.SpyInstance } = {};

  async function runInfoCommand() {
    // eslint-disable-next-line @typescript-eslint/no-var-requires,global-require
    const infoCommand = require('../../src/commands/info');
    await infoCommand.default.run();
  }

  function verifyInfo() {
    verifyStdOut(mocks.writeLine);
  }

  beforeEach(() => {
    resetAll(mocks);

    Object.assign(mocks, mockStdOut());
  });

  afterEach(() => {
    resetMocks(mocks);
  });

  test(`should work`, async () => {
    await runInfoCommand();
    verifyInfo();
  });
});
