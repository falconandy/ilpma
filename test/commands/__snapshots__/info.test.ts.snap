// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'info' should work: writeLine 1`] = `
Array [
  Array [
    "Info works 6",
  ],
]
`;
